`azurerm_windows_web_app` - fix `health_check_eviction_time_in_min` not being set issue

xiaxyi opened the pull request:
This pr is fixing two issues:
1. The property `WEBSITE_HEALTHCHECK_MAXPINGFAILURES` in app_setting is suppose to be set if user assign a value to the property `health_check_eviction_time_in_min`.
2. The `WEBSITE_HEALTHCHECK_MAXPINGFAILURES` should be written back to the state file if user explicitly specify it in config.

fix issues:https://github.com/hashicorp/terraform-provider-azurerm/issues/17123
https://github.com/hashicorp/terraform-provider-azurerm/issues/17546
https://github.com/hashicorp/terraform-provider-azurerm/issues/16302

jackofallops reviewed (changes requested):
Hi @xiaxyi - Thanks for this PR. I've left a couple comments below for changes I believe to be needed. But otherwise I think it looks good 👍 

Thanks!

xiaxyi commented on internal/services/appservice/helpers/web_app_schema.go:
```diff
@@ -3785,8 +3785,8 @@ func ExpandAppSettingsForCreate(settings map[string]string) *[]web.NameValuePair
 
-func FlattenAppSettings(input web.StringDictionary) (map[string]string, *int) {
-	maxPingFailures := "WEBSITE_HEALTHCHECK_MAXPINGFAILURE"
+func FlattenAppSettings(input web.StringDictionary, d sdk.ResourceMetaData) (map[string]string, *int) {
+	maxPingFailures := "WEBSITE_HEALTHCHECK_MAXPINGFAILURES"
```
I think we need to move the `WEBSITE_HEALTHCHECK_MAXPINGFAILURES` from the `unmanagedSettings` list if user adds the key explicitly in the config?

xiaxyi commented on internal/services/appservice/helpers/web_app_schema.go:
```diff
@@ -3803,6 +3803,12 @@ func FlattenAppSettings(input web.StringDictionary) (map[string]string, *int) {
 		healthCheckCount = &h
 	}
 
+	if userInputAppSetting := d.ResourceData.Get("app_settings").(map[string]interface{}); userInputAppSetting != nil {
+		if _, ok := userInputAppSetting[maxPingFailures]; ok {
+			unmanagedSettings = unmanagedSettings[:len(unmanagedSettings)-1]
+		}
+	}
+
```
But azure provided such way to set this property and lots of users are aware of this usage as well. shall we just let them set it via app_setting?

katbyte replied:
we have moved it out of the map into its own property, therefor it should only be set there as @jackofallops, and if the user does put it into the app settings map we should maybe toss up an error telling the user to set it there?

cmoetzing replied:
@katbyte If think the error is really important. I fixed this issue by placing the env variable in `app_settings` and probably many others will have too.

xiaxyi commented:
Thank @jackofallops for the comments, left some feedbacks inline

katbyte commented:
also we have some test failureS:
<img width="399" alt="image" src="https://user-images.githubusercontent.com/1638467/183666762-a1e0eed6-cf3b-4bbf-bf5d-5b788a623b7b.png">

xiaxyi commented:
Thanks @jackofallops for the comments. I removed the app_setting KVP code change, also, the update check of app_setting should be removed as we are collecting data from site_config, which means, app_setting still needs to be updated even there is no change in the config.

For the test cases, I removed the property checks as we have the import step.

xiaxyi commented:
> we still have test failures <img alt="image" width="1220" src="https://user-images.githubusercontent.com/1638467/188040982-45e6fd0f-2c72-4813-b173-fe3f00ecf782.png">

Thanks @katbyte for the comment, the acc test failure will be fixed in https://github.com/hashicorp/terraform-provider-azurerm/pull/18177

pierluca commented:
Is there any ETA on when this issue will be fixed/merged/released ?

jackofallops commented:
Closing as superseded by 19685
